### PR TITLE
Expand simple logging options

### PR DESCRIPTION
**Pull request by Gobberwart**
## Short Description:
Enhancements to logging output options

Adds new "logging" category to config, with the following options:
- color: t/f (replaces existing "colored_logging" setting)
- show_datetime: t/f
- show_process_name: t/f
- show_log_level

Note that this only takes effect after initial output due to the order in which things are loaded. Too much work at this stage to move everything around just for a few lines.

## Fixes/Resolves/Closes (please use correct syntax):
- Replaces PR #4821 
- Expands on #3706

### CODE REVIEW

**Comment by Gobberwart** [comment]
@mjmadsen Is this the sort of thing you had in mind? Feedback VERY MUCH appreciated.

**Comment by Gobberwart** [comment]
Other things we could do with this that I haven't done anything about because I don't know if anyone cares.

- Separate Date/Time into separate values
- Allow users to specify datetime format

**Comment by mjmadsen** [comment]
I'll take a look in the next few hours. Sure I'll love it!

File: pokemongo_bot/__init__.py [hunk 1]
```diff
@@ -760,8 +760,18 @@ def _setup_logging(self):
         logging.getLogger("pgoapi").setLevel(log_level)
         logging.getLogger("rpc_api").setLevel(log_level)
 
-        if self.config.logging_clean and not self.config.debug:
-            formatter = Formatter(fmt='[%(asctime)s] %(message)s', datefmt='%H:%M:%S')
+        if self.config.logging:
+            logging_format = '%(message)s'
+            logging_format_options = ''
+            if ('show_log_level' in self.config.logging and self.config.logging['show_log_level']) or 'show_log_level' not in self.config.logging:
```
Comment (mjmadsen) [inline pokemongo_bot/__init__.py]: Can we check if 'show_log_level' is not present, and if so, set it to true. Then a second if for the actual format change? 

Same with those below.

Comment (Gobberwart) [inline pokemongo_bot/__init__.py]: That's what it does. First check is for "is in config and set true", second check is OR "not in config"

Comment (mjmadsen) [inline pokemongo_bot/__init__.py]: I get that. I'm a bit curious on the evaluation order in python as the "and self.config.logging['show_log_level'] might happen before we check if it is in self.config.logging.

Comment (Gobberwart) [inline pokemongo_bot/__init__.py]: It shouldn't happen... the first condition (with the end) will fail on the first check and not even bother to perform the second.

For the sake of clarity, I'll reverse it :)

Comment (mjmadsen) [inline pokemongo_bot/__init__.py]: It's subtle. 
('show_log_level' in self.config.logging and self.config.logging['show_log_level'])

Might evaluate the latter half first, which would throw an error if it doesn't exist. 

How about this:
if ('show_log_level' not in self.config.logging) or self.config.logging['show_log_level']):

Comment (Gobberwart) [inline pokemongo_bot/__init__.py]: @mjmadsen How's this...?

```
            if 'show_log_level' not in self.config.logging or \
                    ('show_log_level' in self.config.logging and self.config.logging['show_log_level']):
                logging_format = '[%(levelname)s] ' + logging_format
```

**Comment by solderzzc** [comment]
Test Pull approve. 👍

**Comment by solderzzc** [comment]
👍

**Comment by solderzzc** [comment]
@mjmadsen The pull approve is working, you can use 👍  or click on the detail link to set it up.

File: pokemongo_bot/__init__.py [hunk 2]
```diff
@@ -763,11 +763,15 @@ def _setup_logging(self):
         if self.config.logging:
             logging_format = '%(message)s'
             logging_format_options = ''
-            if ('show_log_level' in self.config.logging and self.config.logging['show_log_level']) or 'show_log_level' not in self.config.logging:
+            
+            if 'show_log_level' not in self.config.logging or \
+                    ('show_log_level' in self.config.logging and self.config.logging['show_log_level']):
```
Comment (mjmadsen) [inline pokemongo_bot/__init__.py]: Sorry, I know I'm being a nazi here. The checks in the parenthesis will evaluate first, as they have a higher evaluation precedence, even though python evaluates left to right. "in" has a lower precedence than attributes and indices. 

Please use the following (reduces the double "in" check too:
if ('show_log_level' not in self.config.logging) or self.config.logging['show_log_level']:

File: pokemongo_bot/__init__.py (hunk 1 above)
Comment (Gobberwart) [inline pokemongo_bot/__init__.py]: Hahaha, I'm a derp. If the ('show_log_level' in self.config.logging) fails, I really don't need to check it again, do I? 

2 minutes.

Comment (Gobberwart) [inline pokemongo_bot/__init__.py]: OK, done and tested. All lines now follow the format you suggested as follows:

```
            if ('show_log_level' not in self.config.logging) or self.config.logging['show_log_level']:
                logging_format = '[%(levelname)s] ' + logging_format
```

I knew something wasn't quite right in there.

**Comment by Gobberwart** [comment]
All feedback incorporated. Should be ok to merge?

File: pokemongo_bot/__init__.py (hunk 2 above)
Comment (Gobberwart) [inline pokemongo_bot/__init__.py]: Done

**Comment by mjmadsen** [comment]
@Gobberwart Thank you. Sorry I'm being nuts. It was the double `in` that first caught my eye then my brain kept going :P

**Comment by Gobberwart** [comment]
Heh @mjmadsen all good, I needed that second set of eyes. Better safe than sorry.

**Comment by mjmadsen** [comment]
@solderzzc Yeah, but sometimes CI can pass and we get no attrib errors later. :)

**Comment by mjmadsen** [comment]
@Gobberwart rebase and I'll merge (appears to be out of date).

**Comment by solderzzc** [comment]
@mjmadsen  Test scripts existing in CI, better than nothing... much better.

**Comment by Gobberwart** [comment]
@mjmadsen done

**Comment by solderzzc** [comment]
please try the pull approve :)

**Comment by mjmadsen** [comment]
:+1:

**Comment by mjmadsen** [comment]
:+1:

**Comment by mjmadsen** [comment]
Oh, neat.

**Comment by solderzzc** [comment]
Finally see the 2 green status.

**Pull request merged**